### PR TITLE
HLG-45: Holograph x Doppler Fee Integration

### PR DESCRIPTION
**Pull request by alexanderattar**
# Holograph × Doppler — Fee-Router refactor & cross-chain fee pipeline

Doppler trading fees need to move from Base to Ethereum where they’re converted to HLG and routed into burn + staking logic.  The previous two-slice model and scattered helpers made the flow hard to reason about and expensive on gas.

This PR introduces a simpler **single-slice** model and wires the whole path end-to-end.

### FeeRouter
| Area | Old | New |
|------|-----|-----|
| Split helper | `_takeAndSlice` + extra helpers | `_splitFee` (pure) |
| Conversion   | `_processProtocolFeeLocal*`      | `_convertToHLG` (one path) |
| Burn / stake | `_burnAndStake`                  | `_distribute` (old name kept as shim) |
| ERC-20 bridge| `bridgeToken`                    | `bridgeERC20` (includes amount) |
| LZ decode    | external `try/catch`             | `message.length` check |
| Pool check   | `try/catch` on router            | Cached `IUniswapV3Factory` |

### HolographFactory
* Sends launch ETH straight to `FeeRouter.receiveFee`.
* Writes `integrator = FeeRouter` in `CreateParams`.

### Interfaces / scripts
* `IFeeRouter` exposes `bridgeERC20`.
* Keeper script updated to call new ABI.

## How the fee moves
```
Trader swap → Doppler feesAccrued
             ▼ migrate()
          Airlock.integratorFees
             ▼ keeper pull
        FeeRouter(Base) —— _splitFee()
          │98.5 %              │1.5 %
          ▼                    ▼ LayerZero v2
       Treasury         FeeRouter(Eth) —— _convertToHLG() —— burn & stake
```

## Safety notes
* `KEEPER_ROLE` gates pull + bridge.
* ReentrancyGuard on all externals.
* Trusted remote list per chain.
* `MIN_BRIDGE_VALUE = 0.01 ether` to avoid dust.
* `pause()` stops intake and bridging.

## Tests
* Integration: 9/9
* Unit: 23/23
* StakingRewards: 6/6
* Factory: 1/1  
Everything passes; fuzz tests cover edge cases.

## Deploy plan
1. Deploy FeeRouter on **Base**: `constructor(endpointBase, 30101, 0x0, 0x0, 0x0, 0x0, treasury)`
2. Deploy FeeRouter on **Ethereum**: `constructor(endpointEth, 30184, staking, HLG, WETH, uniRouter, treasury)`
3. Grant `KEEPER_ROLE` to the keeper address on both routers.
4. Set `trustedRemotes` (both directions).
5. Point `StakingRewards.feeRouter` to the Ethereum router.
6. Update keeper env to use `bridgeERC20`.

## Things to look at
* Fee math (`HOLO_FEE_BPS = 150`) and 50/50 burn-stake split.
* `_convertToHLG` route selection and `minOut` handling.
* LayerZero option bytes built by `_buildLzReceiveOption`.
* Updated admin guide in `docs/fee-collection-admin-guide.md`.

---

Files touched: `FeeRouter.sol`, `IFeeRouter.sol`, `HolographFactory.sol`, keeper script, tests, docs.

### CODE REVIEW

File: docs/fee-collection-admin-guide.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+| -------------------------- | -------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **1. Accrual**             | Every swap during an active auction    | Fee deltas recorded in `Doppler.state.feesAccrued` (no transfer yet)                                                                                                                                                                                                                                             |
+| **2. Migration**           | Once sale ends or proceeds cap reached | `Airlock.migrate()` empties liquidity, calls `_handleFees()` twice (token0 & token1). Calculation:<br>`protocolLp = fees / 20` (5 %)<br>`protocolProceeds = (balance-fees) / 1000` (0.1 % proceeds)<br>`protocolFees = max(protocolLp, protocolProceeds)` capped at `fees/5` (20 %). Remaining = integratorFees. |
```
[the remark this answers is not in the document]
Comment (alexanderattar) [reply docs/fee-collection-admin-guide.md:55]: Short answer is basically: `totalFees = protocolFees + integratorFees`

but this this is how the fees are broken down from Doppler's Airlock down to within our contracts:

```
Total Doppler Trading Fees (100%)
         ↓
    Airlock._handleFees()
    ├── Protocol (Doppler): 5-20%
    └── Integrator (FeeRouter): 80-95%  ← This is your `amt` parameter
              ↓
         FeeRouter._splitFee()
         ├── Protocol (Holograph): 1.5% of integrator share
         └── Treasury: 98.5% of integrator share
```

File: docs/holograph_doppler_fee_integration.md [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+   * [Coverage targets & fork commands](#coverage-targets--fork-commands)  
+7. [Migration & Deployment Guide](#migration--deployment-guide)  
+8. [Open Questions & Risks](#open-questions--risks)  
+
+---
+
+## Executive Summary
+We adopt a **single‑slice model** that better matches Doppler's "integrator‑pull" pattern:  
+
+* **HolographFactory** forwards *all* launch ETH to `FeeRouter.receiveFee()` and always sets `integrator = FeeRouter`.
```
[the remark this answers is not in the document]
Comment (alexanderattar) [reply docs/holograph_doppler_fee_integration.md:28]: You bring up a good point. I'm addressing this @sogoiii
